Projects/UOContent/Multis/Boats/BaseBoat.cs

```diff
@@ -1650,13 +1678,37 @@ public bool Move(Direction dir, int speed, int clientSpeed, bool message)
                 return false;
             }
 
+            var verticalMove = false;
+            if(IgnoreWater) {
+                newComponents = MultiData.GetComponents(ItemID);
+                var highZ = 0;
+                for (var x = 0; x < newComponents.Width; ++x)
+                {
+                    for (var y = 0; y < newComponents.Height; ++y)
+                    {
+                        var tx = newComponents.Min.X + x;
+                        var ty = newComponents.Min.Y + y;
+                        var mz = map.Tiles.GetLandTile(tx, ty).Z;
+                        if(highZ < mz) highZ = mz;
+                    }
+                }
+                //very jerky ascent/descent, will try to improve later
+                // if(Flies && Flying) highZ = Math.Min(Z+5,highZ+50);
+                // else if(Z > highZ) highZ = Math.Max(highZ,Z-5);
+                if(Flies) highZ+=50;
```

Comment:
Maybe we want to not introduce this until we have better code around how flying would work?